test/groovy/util/JenkinsStepRule.groovy

```diff
@@ -8,23 +8,49 @@ import org.junit.rules.TestRule
 import org.junit.runner.Description
 import org.junit.runners.model.Statement
 
+
 class JenkinsStepRule implements TestRule {
     final BasePipelineTest testInstance
-
+    final String stepName
     def step
+    def callsIndex = 0
+    def callsParameters = [:]
 
     JenkinsStepRule(BasePipelineTest testInstance) {
         this.testInstance = testInstance
     }
 
+    JenkinsStepRule(BasePipelineTest testInstance, String stepName) {
```

Comment:
Not sure if we need to extend the step rule for catching calls to other steps like this. Why not simply using `registerAllowedMethod(<ourStep>)`?

That would help to keep the focus of that rule clear and simple.

@CCFenner : since you provided that step. What do you think about the enhancement here?

Reply:
I will move it to a new class JenkinsMockStepRule.